### ISSUE TITLE
ENA-198: Using keyboard to navigate to the Sounds tab results in a crash

### ISSUE THREAD

**Issue by mxmou** [LLK/scratch-gui#8545]
### Expected Behavior

If something can be done using a keyboard, it would behave the same way as when used with a mouse.

### Actual Behavior

Using only a keyboard (no mouse clicks) to move to the Sounds tab crashes with the following error message: `TypeError: this.audioContext.createBuffer is not a function`

### Steps to Reproduce

1. Open the editor.
2. Press Tab until the tabs are focused. Don't click anything in the editor.
3. Use arrow keys to select the Sounds tab.

### System Details

Windows 11. Happens in both Edge 105 and Scratch app 3.29.1.

**Comment by cwillisf** [comment]
Good catch! I suspect this is because, in `src/lib/audio/shared-audio-context.js`, we only watch for `mousedown` or `touchstart` and not any keyboard events. If that's the problem, then causing even a single touch or mouse click would prevent this crash. Can you confirm that's what you're seeing? Based on "Don't click anything in the editor" I think it probably is... :)

**Comment by mxmou** [comment]
> If that's the problem, then causing even a single touch or mouse click would prevent this crash. Can you confirm that's what you're seeing? Based on "Don't click anything in the editor" I think it probably is... :)

Yes, any mouse click prevents the crash.

**Comment by travissanderson** [comment]
@mxmou can you still reproduce this issue? I thought I might pick this up, but I'm not able to tab to the Sound tab at all. Looks like there's a PR open to address the issue if you are still able to reproduce it https://github.com/LLK/scratch-gui/pull/8570

**Comment by mxmou** [comment]
@travissanderson Navigating using the keyboard is difficult because most elements don't have a visible focus state, but it's possible and I can still reproduce the bug. When the Code tab has focus, arrow keys can be used to switch to a different tab.

**Comment by travissanderson** [comment]
Hmm, on Mac in Chrome after I tab thru the sprites and backdrop controls and such, it goes to the controls on the radial direction control and after that goes to the browser UI instead. As far as accessibility goes, how do you even tell in your setup that the Code tab has focus? It might be prudent to enable some selected UI at least if the keyboard is being used to tab through elements, that would be a separate issue from this one but closely related. What do you think?

**Comment by CST1229** [comment]
> @mxmou can you still reproduce this issue? I thought I might pick this up, but I'm not able to tab to the Sound tab at all. Looks like there's a PR open to address the issue if you are still able to reproduce it #8570

I can reproduce by waiting for the editor to load, then pressing Tab three times (to focus the tabs area) and pressing the right arrow key twice (to move to the Sounds tab). If you've focused the extensions button or anything on the sprite pane, then you've already tabbed past the tabs area.

**Comment by mxmou** [comment]
> I can reproduce by waiting for the editor to load, then pressing Tab three times (to focus the tabs area) and pressing the right arrow key twice (to move to the Sounds tab). If you've focused the extensions button or anything on the sprite pane, then you've already tabbed past the tabs area.

The number of tabs required also depends on whether the user is logged in.

**Comment by travissanderson** [comment]
Thanks, guys, I'll take another look here in the next week or so. @CST1229 is there a reason we don't want to move your open PR forward to resolve the bug?

**Comment by CST1229** [comment]
> @CST1229 is there a reason we don't want to move your open PR forward to resolve the bug?

I don't know, you [the Scratch developers] decide if pull requests actually get merged or not, not me. If I didn't want it to get merged, I would just close it or not make it at all.

**Comment by travissanderson** [comment]
Understood! I am just here looking to maybe contribute and was curious about the status. Thanks :)
